bokeh/_templates/autoload_js.js

```diff
@@ -26,31 +26,38 @@ calls it with the rendered model.
   if (typeof (window._bokeh_onload_callbacks) === "undefined"){
     window._bokeh_onload_callbacks = [];
   }
-  function load_lib(url, callback){
+  function load_libs(js_urls, callback) {
     window._bokeh_onload_callbacks.push(callback);
-    if (window._bokeh_is_loading){
+    if (window._bokeh_is_loading > 0) {
       console.log("Bokeh: BokehJS is being loaded, scheduling callback at", new Date());
       return null;
     }
     console.log("Bokeh: BokehJS not loaded, scheduling load and callback at", new Date());
-    window._bokeh_is_loading = true;
-    var s = document.createElement('script');
-    s.src = url;
-    s.async = true;
-    s.onreadystatechange = s.onload = function(){
-      {%- for file in css_files %}
-      Bokeh.embed.inject_css("{{ file }}");
-      {%- endfor %}
-      window._bokeh_onload_callbacks.forEach(function(callback){callback()});
-    };
-    s.onerror = function(){
-      console.warn("failed to load library " + url);
-    };
-    document.getElementsByTagName("head")[0].appendChild(s);
+    window._bokeh_is_loading = js_urls.length;
+    for (i = 0; i < js_urls.length; i++) {
+      url = js_urls[i];
+      var s = document.createElement('script');
+      s.src = url;
+      s.async = true;
+      s.onreadystatechange = s.onload = function() {
+        window._bokeh_is_loading--;
+        if (window._bokeh_is_loading==0) {
+          console.log("Bokeh: all BokehJS libraries loaded");
+          {%- for file in css_files %}
+          console.log("Bokeh: injecting CSS: {{ file }}");
+          Bokeh.embed.inject_css("{{ file }}");
+          {%- endfor %}
+          window._bokeh_onload_callbacks.forEach(function(callback){callback()});
+        }
+      };
+      s.onerror = function() {
+        console.warn("failed to load library " + url);
+      };
+      console.log("Bokeh: injecting script tag for BokehJS library: ", url);
+      document.getElementsByTagName("head")[0].appendChild(s);
```

Comment:
Apparently we aren't guaranteed that these scripts will run in the right order; http://stackoverflow.com/a/3231664 talks about it, in a thing linked from there someone claims IE will execute them out of order and firefox/chrome won't. I think that means we could run bokeh-widgets.js when bokeh.js has not been run yet.

To solve it I guess we could manually chain (in each callback, add the next script tag), though that will prevent them from downloading in parallel... or we could remove async=true, which is maybe the same thing... or long-run (maybe not today) we could maybe only load a require.js setup "by hand" here, and then use require() for everything else...